---
title: "RNAseq LMB2"
subtitle: "Thoenen_et_al_BX_metabolisation"
author: "Lisa Thoenen"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
rm(list = ls())
```

## Set settings
set working directory to source file location

## Packages
set working directory to source file location
```{r, warning=FALSE, include=FALSE}
# install.packages("MESS")
# #Plotly to annotate plots and explore them
# #install.packages("plotly")
# install.packages("metacoder")
# 
# #lme4 and lmertest for stats, multcomp and lsmeans for posthoc
# install.packages("lme4")
# install.packages("lmerTest")
# install.packages("multcomp")
# install.packages("emmeans")
# install.packages("magrittr")
# install.packages("readxl")
# install.packages("lme4")
# install.packages("lmerTest")
# install.packages("multcomp")
# install.packages("emmeans")
# 
# #Formatting and reading tools
# install.packages("magrittr")
# install.packages("tidyverse")
# install.packages("readxl")
# install.packages("lubridate")
# install.packages("broom")
# install.packages("dplyr")
# install.packages("tidyr")
# install.packages("stringr")
# install.packages("readr")
# install.packages("purrr")
# 
# #Plotstuff
# install.packages("ggthemes")
# install.packages("ggforce")
# install.packages("KEGGREST")
# install.packages("randomForest")
# install.packages("ggplot2")
# install.packages("rstatix")

# install.packages("pander")
# install.packages("ggtree")
# install.packages("tibble")
# install.packages("knitr")
```

##Library loading, setting up misc functions
```{r, warning=FALSE, include=FALSE}
#MESS for AUC function
library("MESS")
#Plotly to annotate plots and explore them
#library("plotly")

library("metacoder")

#lme4 and lmertest for stats, multcomp and lsmeans for posthoc
library("lme4")
library("lmerTest")
library("multcomp")
library("emmeans")
#Formatting and reading tools
library("magrittr")
#library("tidyverse")
library("readxl")
library("lubridate")
library("broom")
library("dplyr")
library("tidyr")
library("stringr")
library("readr")
library("purrr")
library("rstatix")
### the dplyr package gets overwritten by other packages so often it is necessary to load it specifically again when needed for exapmle like that: dplyr::select
#Plotstuff
library("ggthemes")
library("ggforce")
#library("KEGGREST")
#library("randomForest")
library("ggpubr")
library("pander")
library("ggtree")
library("tibble")
library("knitr")

```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE}
genes <- read.csv("Input/condition_MBOA_vs_control_fun.csv", header = TRUE)

genes$location <- genes$gene

genes$location <- gsub("LMB2_", "", genes$location)
genes %<>% mutate(location = as.numeric(location))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE}
genes_sig <- genes %>% filter(padj < 0.05)
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE}
write.csv(genes_sig, "Input/genes_sig_RNAseq.csv")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE}
genes_sig_cluster <- genes_sig %>% filter(gene %in% c("LMB2_002997", "LMB2_002998", "LMB2_002999", 
"LMB2_003000",
"LMB2_003001",
"LMB2_003002",
"LMB2_003003",
"LMB2_003005",
"LMB2_003006",
"LMB2_003007",
"LMB2_003008",
"LMB2_003010",
"LMB2_003011",
"LMB2_003012",
"LMB2_003013"))

genes_cluster <- genes %>% filter(gene %in% c("LMB2_002997", "LMB2_002998", "LMB2_002999", 
"LMB2_003000",
"LMB2_003001",
"LMB2_003002",
"LMB2_003003",
"LMB2_003005",
"LMB2_003006",
"LMB2_003007",
"LMB2_003008",
"LMB2_003010",
"LMB2_003011",
"LMB2_003012",
"LMB2_003013"))
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include=FALSE}
write.csv(genes_sig_cluster, "Input/genes_sig_cluster_RNAseq.csv")
```

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include=FALSE}
genes_sig$gene %>% length()

genes_sig_pos <- genes_sig %>% filter(log2FoldChange > 0)
genes_sig_pos$gene %>% length()
```

- 108 genes are significantly differentially expressed
- 94 of them are upregulated.

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include=FALSE}
genes_sig_pos %<>% mutate(cluster = case_when(gene %in% genes_cluster$gene ~ "cluster",
                                             !gene %in% genes_cluster$gene ~ "not cluster",))
```


```{r, message=FALSE, echo = FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
genes_reg_cluster <- genes_sig_pos %>% ggplot(aes(x = gene, y = location)) + 
  geom_point(aes(colour = log2FoldChange)) +
  scale_colour_gradient(low = "gold", high = "darkred") +
  theme_bw() +
  theme(axis.text.x=element_blank())

genes_reg_cluster

# ggsave(plot = genes_reg_cluster,  filename = "genes_reg_cluster.pdf", width = 17, height = 15, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = genes_reg_cluster,  filename = "genes_reg_cluster.svg", width = 17, height = 15, dpi = 300, scale = 1, units = "cm")
```

```{r}
sessionInfo()
```

