---
title: "RNAseq LMB2"
subtitle: "Thoenen et al. AMPO formation"
author: "Lisa Thoenen & Christine Pestalozzi"
date: "`r Sys.Date()`"
output: 
  html_document:
    fig_caption: yes
    toc: true
    toc_float: true
    theme: cerulean
    code_folding: show
editor_options: 
  chunk_output_type: console
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, include=FALSE}
rm(list = ls())
```

## Set settings
set working directory to source file location

## Packages
```{r, warning=FALSE, include=FALSE}
# install.packages("tidyverse")
# install.packages("magrittr")
# install.packages("readxl")
# install.packages("ggpubr")
# install.packages("knitr")
```

## Library loading, setting up misc functions
```{r, warning=FALSE, include=FALSE}
library("tidyverse")
library("magrittr")
library("readxl")
library("ggpubr")
library("knitr")

```


# read in data
Read in RNAseq output from DESeq2 and list of bxd cluster genes.

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include=FALSE}
#RNAseq output
genes <- readxl::read_xlsx("Input/DatasetS4_RNAseq_LMB2_corrected.xlsx", na = 'NA')

genes$location <- genes$gene
genes$location <- gsub("LMB2-1.2_", "", genes$location)
genes %<>% mutate(location = as.numeric(location))

#bxd gene cluster
bxdcluster <- read.csv("Input/bxd_cluster.csv", header = T)

```

Filter significantly regulated genes (based on BH-corrected P value < 0.05)
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include=FALSE}

genes_sig <- genes %>% filter(padj < 0.05)
genes_sig_cluster <- genes_sig %>% filter(gene %in% bxdcluster$Gene_Identifier)

genes_cluster <- genes %>% filter(gene %in% bxdcluster$Gene_Identifier)

genes_sig$gene %>% length()

genes_sig_pos <- genes_sig %>% filter(log2FoldChange > 0)
genes_sig_pos$gene %>% length()
```

- `r genes_sig$gene %>% length()` genes are significantly differentially expressed
- `r genes_sig_pos$gene %>% length()` of them are upregulated.

```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include=FALSE}
genes_sig_pos %<>% mutate(cluster = case_when(gene %in% genes_cluster$gene ~ "cluster",
                                             !gene %in% genes_cluster$gene ~ "not cluster",))
```

Read in the kmer data from comparative genomics done on selected Microbacterium genomes. 
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include=FALSE}
kmer_sig <- read_excel("Input/DatasetS3_kmer_LMB2_corrected.xlsx")
genes_kmer <- kmer_sig$Gene

```

group genes into RNAseq significant and kmer signal significant 
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE, include=FALSE}
significant <- setdiff(genes_sig$gene, genes_kmer)
significantKmer <- intersect(genes_sig$gene, genes_kmer)
kmeronly <- setdiff(genes_kmer, genes_sig$gene)

# add the grouping to the RNAseq
genes$Group <- "not significant"
genes$Group[genes$gene %in% significant] <- "significant"
genes$Group[genes$gene %in% kmeronly] <- "not significant & kmer signal"
genes$Group[genes$gene %in% significantKmer] <- "significant & kmer signal"

genes$Group <- factor(genes$Group, levels = c("significant & kmer signal",
                                              "significant",
                                              "not significant",
                                              "not significant & kmer signal"))
```

# Plot 

Fig S9c and S9d
S9c) adjusted p-value vs log2FC: only genes who could be assessed (i.e. padj not NA) are shown. 
S9d) mean gene expression vs log2FC: genes with mean expression above 25000are removed for better visibility (rRNAs, tRNA)
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE}
panel_c <- ggplot(genes, aes(x = log2FoldChange, y = -log(padj)))+
  geom_point(aes(color = Group))+
  scale_color_manual(values = c("blue","red", "grey","lightblue"))+
  labs(title = "Gene expression MBOA - DMSO",
       x = expression(paste(log[2], "FC", sep = "")))

panel_c

panel_d_sub <-  ggplot(genes, aes(x = log2FoldChange, y = baseMean))+
  geom_point(aes(color = Group))+
  scale_color_manual(values = c("blue","red", "grey","lightblue"))+
  scale_y_continuous(limits = c(0,25000))+
  labs(title = "Mean gene expression",
       x = expression(paste(log[2], "FC", sep = "")))

panel_d_sub

panelcd <- ggarrange(panel_c, panel_d_sub, nrow = 2, common.legend = T, align = "v", labels = c("c", "d"), legend = "bottom")

genes %>% group_by(Group) %>% summarise(n = n())
```


Fig S9e: Plot with upregulated genes marked along the genome
```{r, message=FALSE, echo = FALSE, error=FALSE, warning = FALSE, fig.dim = c(6, 5)}
bxd_cluster_sorted <- genes_cluster %>% dplyr::arrange(location)
padding <- 100

genes_reg_cluster <- genes_sig_pos %>% 
  ggplot(aes(x = gene, y = location)) + 
  geom_point(aes(colour = log2FoldChange)) +
  scale_colour_gradient(low = "gold", high = "darkred") +
  theme_bw() +
  theme(axis.text.x=element_blank())+
  annotate("rect", xmin = which(levels(factor(genes_sig_pos$gene)) == bxd_cluster_sorted$gene %>% min()) -0.5, 
           xmax = which(levels(factor(genes_sig_pos$gene)) == bxd_cluster_sorted$gene %>% max()) +0.5, 
           ymin = bxd_cluster_sorted$location %>% min() -padding , ymax = bxd_cluster_sorted$location %>% max() +padding,
           alpha = .1, color = "black", fill = NA, linetype = "dotted", linewidth = 1)+
  labs(title = "Upregulated genes along the genome",   
       y = "Location", 
       x = "Upregulated gene",
       color = expression(paste(log[2], "FC", sep = "")))+
    theme(legend.position = c(0.8, 0.2))+
    annotate("text", x = which(levels(factor(genes_sig_pos$gene)) == bxd_cluster_sorted$gene %>% min()) -0.5, 
             y = bxd_cluster_sorted$location %>% max() +padding +100 ,
             label = expression(italic("bxdA")~"cluster"))

genes_reg_cluster

```

Combine panels Fig S9c to Fig S9e and export as one Figure
```{r, message=FALSE, echo=FALSE, error=FALSE, warning = FALSE}
panels_c_to_e <- ggarrange(panelcd, genes_reg_cluster, ncol = 2, labels = c("", "e"), align = "v")

ggsave(plot = panels_c_to_e, filename = "FigS9c_to_e.svg", width = 17, height = 11, dpi = 300, scale = 2, units = "cm")

```

```{r}
sessionInfo()
```

