---
title: "Comparison lactonase bxdA"
subtitle: "Thoenen_et_al_Bacteria_BX_metabolization"
author: "Lisa Th√∂nen"
date: "`r Sys.Date()`"
output: 
  html_document:
    fig_caption: yes
    toc: true
    toc_float: true
    theme: cerulean
    code_folding: show
editor_options: 
  chunk_output_type: console
---

# *bxdA* in other strains
locus tag: LMB2-1.2_002078 genome: LMB2-1.2
species: Microbacterium taxid: 33882
KEGG gene name: *actIV*
on amino acid level (AA)

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, include=FALSE}
rm(list = ls())
```

```{r, include=FALSE}
library(magrittr)
# library(tidyverse)
library(pheatmap)
library(readxl)

#MESS for AUC function
library("MESS")
#Plotly to annotate plots and explore them
#library("plotly")

library("metacoder")

#lme4 and lmertest for stats, multcomp and lsmeans for posthoc
library("lme4")
library("lmerTest")
library("multcomp")
library("emmeans")
#Formatting and reading tools
library("magrittr")
#library("tidyverse")
library("readxl")
library("lubridate")
library("broom")
library("dplyr")
library("tidyr")
library("stringr")
library("readr")
library("purrr")
library("rstatix")
### the dplyr package gets overwritten by other packages so often it is necessary to load it specifically again when needed for exapmle like that: dplyr::select
#Plotstuff
library("ggthemes")
library("ggforce")
#library("KEGGREST")
#library("randomForest")
library("ggpubr")
library("pander")
library("ggtree")
library("tibble")
```

# Comparison among MRB

```{r, message=F, echo=F, warning=F}
identity_AA_MRB <- read.table("Input/actIV_similarity_selected_strains_taxa.txt")
```

```{r, message=F, echo=F, warning=F}
similarity_lactonase_MRB <- identity_AA_MRB %>% filter(!V2 %in% c("LMB21.2_002078", "LMB211.1_002997", "LMS41.2_002646", "LMS411.1_001325")) %>% 
  ggplot(aes(x = 1, y = V2)) + 
  geom_tile(aes(fill = V4)) +
  geom_text(aes(label = V4))+
  theme_bw() +
  scale_fill_gradient2(low = "white", mid = "cornflowerblue", high = "darkred", midpoint = 35) +
  theme(axis.text.x=element_blank())+
  labs(y = "",
       x = " ",
       fill = "identity [%]")

similarity_lactonase_MRB

# ggsave(plot = similarity_lactonase_MRB, filename = "similarity_lactonase_MRB.pdf", width = 8.8, height = 9, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = similarity_lactonase_MRB, filename = "similarity_lactonase_MRB.svg", width = 8.8, height = 9, dpi = 300, scale = 1, units = "cm")
```

# Comparison GMGC

```{r, message=F, echo=F, warning=F}
identity_DNA_GMGC <- read_excel("Input/similarity_gmgc.xlsx")
```

```{r, message=F, echo=F, warning=F}
similarity_lactonase_GMGC <- identity_DNA_GMGC %>% 
  ggplot(aes(x = 1, y = Taxon)) + 
  geom_tile(aes(fill = ID)) +
  geom_text(aes(label = ID))+
  theme_bw() +
  scale_fill_gradient2(low = "white", mid = "cornflowerblue", high = "darkred", midpoint = 35) +
  theme(axis.text.x=element_blank())+
  labs(y = "",
       x = " ",
       fill = "identity [%]")

similarity_lactonase_GMGC

# ggsave(plot = similarity_lactonase_GMGC, filename = "similarity_lactonase_GMGC.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = similarity_lactonase_GMGC, filename = "similarity_lactonase_GMGC.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

# Comparison Microbacteria 

```{r, message=F, echo=F, warning=F}
identity_AA_Micro <- read.table("Input/actIV_comparision_AA_Micro.txt")
```

```{r, message=F, echo=F, warning=F}
similarity_lactonase_AA_Micro <- identity_AA_Micro %>% 
  ggplot(aes(x = 1, y = V2)) + 
  geom_tile(aes(fill = V5)) +
  geom_text(aes(label = V5))+
  theme_bw() +
  scale_fill_gradient2(low = "white", mid = "cornflowerblue", high = "darkred", midpoint = 35) +
  theme(axis.text.x=element_blank())+
  labs(y = "",
       x = " ",
       fill = "identity [%]")

similarity_lactonase_AA_Micro

# ggsave(plot = similarity_lactonase_AA_Micro, filename = "similarity_lactonase_AA_Micro.pdf", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = similarity_lactonase_AA_Micro, filename = "similarity_lactonase_AA_Micro.svg", width = 8.8, height = 6, dpi = 300, scale = 1, units = "cm")
```

```{r, , message=F, echo=F, warning=F}
library(magrittr)
library(ggtree)
### Newick tree ###

species_tree <- ape::read.tree("Input/SpeciesTree_rooted.txt")
metadata <- read_excel("Input/211108_expdesign_Microbacteria_MRB_At_tolerance_DMGmet.xlsx", sheet = 2)

metadata_tree <- metadata %>% mutate(Host_plant = Collection) %>% 
  dplyr::rename(Strain = Strain_ID) %>% 
  mutate(Host_plant = gsub("Species_collection", "maize", Host_plant)) %>% 
  mutate(Host_plant = gsub("Maize_collection", "maize", Host_plant)) %>% 
  mutate(Host_plant = gsub("MBOA", "maize", Host_plant)) %>% 
  mutate(Host_plant = gsub("Compartment_collection", "maize", Host_plant)) %>% 
  mutate(Host_plant = gsub("AtSphere", "arabidopsis", Host_plant)) %>% 
  mutate(Host_plant = case_when(Strain == "LBN7" ~ "brassica", Strain != "LBN7" ~ Host_plant)) %>% 
  mutate(Host_plant = case_when(Strain == "LTA6" ~ "wheat", Strain != "LTA6" ~ Host_plant)) %>% 
  mutate(Host_plant = case_when(Strain == "LMS4" ~ "medicago", Strain != "LMS4" ~ Host_plant)) %>% 
  # mutate(Host_plant = case_when(Strain == "KHBO19" ~ "clover", Strain != "KHB019" ~ Host_plant)) %>% 
  dplyr::select(Strain, AMPO_on_MBOA_plates, Host_plant) %>% 
  filter(!Strain %in% NA) %>% 
  filter(!Strain %in% "NA") %>% 
  mutate(AMPO_on_MBOA_plates = gsub("no", "negative", AMPO_on_MBOA_plates)) %>% 
  mutate(AMPO_on_MBOA_plates = gsub("yes", "positive", AMPO_on_MBOA_plates)) %>% 
  add_row(Strain = "KHB019", AMPO_on_MBOA_plates = "negative", Host_plant = "clover") %>%  mutate(Host_plant = factor(Host_plant, levels = c("maize", "arabidopsis", "brassica", "medicago", "wheat", "clover", "NA")))
  
p <-  species_tree %>% 
      ggtree::ggtree(branch.length = "branch.length", ladderize = TRUE)  %<+% metadata_tree + 
      geom_tippoint(aes(color=AMPO_on_MBOA_plates), size = 3) + 
      geom_tiplab(align=TRUE, linetype='dashed', linesize=.3, size = 3, offset = 0.02) + 
      scale_color_manual(values = c("gold", "darkred")) +
      xlim(0, 2) 

metadata_tree_2 <- metadata_tree %>% 
        mutate(Host_plant = gsub("brassica", "other", Host_plant)) %>% 
        mutate(Host_plant = gsub("medicago", "other", Host_plant)) %>% 
        mutate(Host_plant = gsub("wheat", "other", Host_plant)) %>% 
        mutate(Host_plant = gsub("clover", "other", Host_plant))

p_host_simple <-  species_tree %>% 
      ggtree::ggtree(branch.length = "branch.length", ladderize = TRUE) %<+% metadata_tree_2 +
      geom_tippoint(aes(color=Host_plant), size = 3) + 
      geom_tiplab(align=TRUE, linetype='dashed', linesize=.3, size = 3, offset = 0.02) +
      scale_color_manual(values = c("#3e5437", "tan3", "grey"), labels = c("maize", "arabidopsis", "other"), name = "Host plant") +
      xlim(0, 2) 
  
p_host_simple
```

```{r, message=F, echo=F, warning=F}
gene_name <- read_excel("Input/gene_name_lactonase_Micro.xlsx")
identity_AA_Micro_strain <- left_join(identity_AA_Micro %>% filter(!V2 %in% c("LWH71.2_003460")), gene_name, by = c("V2" = "gene"))

identity_AA_Micro_strain_meta <- left_join(metadata, identity_AA_Micro_strain, by = c("Strain_ID" = "Strain"))
```


```{r, message=F, echo=F, warning=F}
similarity_lactonase_AA_Micro_tree <- ggtree::gheatmap(p,
         identity_AA_Micro_strain_meta %>%
         dplyr::select(Strain_ID, V5) %>% unique() %>% 
         as.data.frame() %>% column_to_rownames("Strain_ID") ,
         offset=0.5,
         width=0.5,
         colnames=FALSE,
         font.size = 3,
         colnames_position = "top",
         colnames_offset_y = 5,
         colnames_angle = 90,
         colnames_level = NULL,
         hjust =1) +
  scale_fill_gradient2(low = "white", mid = "cornflowerblue", high = "darkred", midpoint = 35) +
  labs(fill = "identity [%]", color = "AMPO phenotype") 
#dev.off()

similarity_lactonase_AA_Micro_tree

#ggsave(plot = similarity_lactonase_AA_Micro_tree,  filename = "similarity_lactonase_AA_Micro_tree.pdf", width = 15, height = 19, dpi = 300, scale = 1, units = "cm")
# 
#ggsave(plot = similarity_lactonase_AA_Micro_tree,  filename = "similarity_lactonase_AA_Micro_tree.svg", width = 15, height = 19, dpi = 300, scale = 1, units = "cm")
```

# Comparison Blast

```{r, message=F, echo=F, warning=F}

identity_blast <- read_excel("Input/Lactonase_blast_results.xlsx")
```

```{r, message=F, echo=F, warning=F}
similarity_lactonase_blast <- identity_blast %>% 
  mutate(`Per. Ident` = as.numeric(`Per. Ident`)*100) %>% 
  ggplot(aes(x = 1, y = reorder(interaction(`Scientific Name`, Accession), desc(`Scientific Name`)))) + 
  geom_tile(aes(fill = `Per. Ident`)) +
  geom_text(aes(label = `Per. Ident`))+
  theme_bw() +
  scale_fill_gradient2(low = "white", mid = "cornflowerblue", high = "darkred", midpoint = 35) +
  theme(axis.text.x=element_blank())+
  labs(y = "",
       x = " ",
       fill = "identity [%]")

similarity_lactonase_blast

# ggsave(plot = similarity_lactonase_blast, filename = "similarity_lactonase_blast.pdf", width = 15, height = 12, dpi = 300, scale = 1, units = "cm")
# 
# ggsave(plot = similarity_lactonase_blast, filename = "similarity_lactonase_blast.svg",width = 15, height = 12, dpi = 300, scale = 1, units = "cm")
```

```{r}
sessionInfo()
```
