---
title: "Phylogenetic tree maize root bacteria collection AMPO formation"
subtitle: "Thoenen et al. AMPO formation"
author: "Lisa Thoenen, update by Christine Pestalozzi"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_float: true
    theme: cerulean
    code_folding: show
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r,include=FALSE}
rm(list=ls())
```

```{r,include=FALSE}
# install.packages("dplyr")
# install.packages("plyr")
# install.packages("magrittr")
# install.packages("ggplot2")
# install.packages("readxl")
# install.packages("tidyr")
# install.packages("readr")
# install.packages("stringr")
# install.packages("ggplot2")
# install.packages("forcats")
# install.packages("ggpmisc")
# install.packages("pander")
# install.packages("tidyr")
# install.packages("tibble")
# install.packages("RColorBrewer")
# install.packages("pheatmap")
# install.packages("ggdendro")
# install.packages("reshape2")
# install.packages("grid")
# install.packages("ape")
# install.packages("ggtree")
# install.packages("aplot")
# install.packages("wesanderson")
# install.packages("ggsci")
# install.packages("ggnewscale")
# install.packages("data.tree")
# install.packages("networkD3")
# install.packages("Cairo")
# install.packages("phytools")

# install.packages("ggtreeExtra")
# install.packages("treeio")
# 
# install.packages(pckgs)

# ggtree package

# if (!requireNamespace("BiocManager", quietly = TRUE))
#      install.packages("BiocManager")
# 
# BiocManager::install("ggtree")
# BiocManager::install("ggtreeExtra")
```

```{r, include=FALSE}
library(plyr) # please load plyr first, then dplyr
library(dplyr)
library(magrittr)
library(ggplot2)
library(readxl)
library(tidyr)
library(readr)
library(stringr)
library(ggplot2)
library(forcats)
library(ggpmisc)
library(pander)
library(tidyr)
library(tibble)
library(RColorBrewer)
library(pheatmap)
library(ggdendro)
library(reshape2)
library(grid)
library(ape)
library(ggtree)
# library(aplot)
library(wesanderson)
library(ggsci)
library(ggnewscale)
library(data.tree)
library(networkD3)
library(Cairo)
library(phytools)
# library(ggtreeExtra)
# library(treeio)
```


## Load and format input data
Input: *Metadata bacteria:* Database_MRB_isolates & database_raw, *Mapping results:* Mapping in 4 soils and feedback, *Phylogeny:* Tree computed from 16s sequences (Sanger). Remove tips without any experimental data

```{r,include=FALSE}

database_raw <- read_excel("Input/MRB_strains_AMPO_pheno.xlsx")
Database_MRB_isolates <- read.csv("Input/Database_MRB_isolates.csv") 

```

Focus only on strains with AMPO phenotype. So filter the database accordingly
```{r, warning=FALSE, include=FALSE}
database_raw <- left_join(database_raw, Database_MRB_isolates, by = "Strain") #add new names
database_raw_fil <- database_raw %>% dplyr::select(-Strain_old.x) 
```

Read in the 16 Tree. Remove tips of strains without AMPO phenotype. 
```{r, warning=FALSE, include=FALSE}
### Newick tree ###
Tree_16s_unique <- 
   ggtree::read.tree("Input/220427_Tree_Marco_RAxML_bipartitionsBranchLabels.concatenated_rooted")

tips_tree <- Tree_16s_unique$tip.label
if(!all(tips_tree %in% database_raw_fil$Strain)) warning("not all strains in tree have phenotype information") # stop if there are strains in the tree that have no phenotype information

# database_raw_fil$Strain[!(database_raw_fil$Strain %in% tips_tree)] 
# if(any(!(database_raw_fil$Strain %in% tips_tree))){ 
#   cat("Strains present in tree but not in phenotype file:", database_raw_fil$Strain[!(database_raw_fil$Strain %in% tips_tree)])}
# 
# if(any(duplicated(database_raw_fil$Strain))) {
# cat("Strains present more than once in the phenotype file:", database_raw_fil$Strain[duplicated(database_raw_fil$Strain)] )}

# check overlap of tip labels and database_raw strains with phenotype information
Strains_to_show <- sum(Tree_16s_unique$tip.label %in% database_raw_fil$Strain)
cat("Number of strains in tree to show:", Strains_to_show)

Tree_16s_unique <- drop.tip(Tree_16s_unique, tip = Tree_16s_unique$tip.label[!(Tree_16s_unique$tip.label %in% database_raw_fil$Strain)])

database_tree <- database_raw %>%  filter(Strain %in% Tree_16s_unique$tip.label)

```


## Abundance in greenhouse based on mapping of Thoenen et al PNAS 2023 
Hu Feedback cumulative abundance per strain at a 99% similarity cutoff, greenhouse B73 WT
```{r, error = FALSE, echo=FALSE, message=FALSE}
cutoff_similarity <- 0.99

# read in relative abundance data of Hu et al 
ASVs_Hu_Feedback <- read.delim("Input/all_phy_psmelt_FB_BXp_iso_new_mapping.csv", sep = ",", row.names = 1)

# calculate the cumulative relative abundance of ASVs for each strain
ASVs_Hu_Feedback_cum_abund_99 <- ASVs_Hu_Feedback %>% filter(Strain %in% database_tree$Strain) %>% 
  filter(X. > cutoff_similarity) %>% 
  dplyr::select(OTU, Strain, Abundace_mean) %>% 
  unique() %>% # only keep one copy for each ASV 
  group_by( Strain) %>% 
  summarise(Greenhouse_cum_abundance_99 = sum(Abundace_mean))

database <- left_join(database_tree, ASVs_Hu_Feedback_cum_abund_99 %>% ungroup() %>%  select(starts_with("Greenhouse_cum"), Strain), by = c("Strain"))

```
For getting the cumulative abundance of ASVs matching isolates, relative abundances of all ASVs that mapped with >`r 100*cutoff_similarity`% similarity are summed at the level of the isolate. The mapping was done based on 16S Sanger sequences and taken from Thoenen et al 2023 PNAS.


```{r, warning=FALSE, include=FALSE}
# join metadata to the tree
tree_data <- database %>% mutate(id = str_c(Genus, Strain, sep ="_")) 

# filter strains in tree
strains_tree <- Tree_16s_unique$tip.label %>% unique() 
database_raw %>% filter(!Strain %in% strains_tree) %>% unique() 
```


```{r, warning=FALSE, include=FALSE}
database$Genus <- gsub("Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium", "Rhizobium", database$Genus)
database$Genus <- gsub("Micrococcineae", "Microbacterium", database$Genus)
database$Phylum <- gsub("Actinobacteriota", "Actinobacteria", database$Phylum)
database$Phylum <- gsub("Bacteroidota", "Bacteroidetes", database$Phylum)
```

```{r, warning=FALSE, include=FALSE}
# Assignment of colours families

database$cols_Family <- as.character(database$Family)
database$Family <- gsub("Chitinophagales", "Chitinophagaceae", database$Family)
database$Family <- gsub("Pseudomonadales", "Pseudomonadaceae", database$Family)
database[database$Family=="Bacillaceae" , ]$cols_Family <- "#004586"
database[database$Family=="Chitinophagaceae" , ]$cols_Family <- "#ff420e"
database[database$Family=="Enterobacteriaceae" , ]$cols_Family <- "#ffd320"
database[database$Family=="Erwiniaceae" , ]$cols_Family <- "#579d1c"
database[database$Family=="Microbacteriaceae" , ]$cols_Family <- "#7e0021"
database[database$Family=="Micrococcaceae" , ]$cols_Family <- "#83caff"
database[database$Family=="Nocardioidaceae" , ]$cols_Family <- "#314004"
database[database$Family=="Oxalobacteraceae" , ]$cols_Family <- "#aecf00"
database[database$Family=="Pseudomonadaceae" , ]$cols_Family <- "#4b1f6f"
database[database$Family=="Rhizobiaceae", ]$cols_Family <- "#ff950e"
database[database$Family=="Sphingomonadaceae", ]$cols_Family <- "#c5000b"
database[database$Family=="Xanthomonadaceae", ]$cols_Family <- "#0084d1"
database[database$Family=="Streptomycetaceae" , ]$cols_Family <- "#8b995a"
database[database$Family=="Flavobacteriaceae" , ]$cols_Family <- "#F0D5B4"
database[database$Family=="Paenibacillaceae" , ]$cols_Family <- "#8f9ec9"
database[database$Family=="Planococcaceae", ]$cols_Family <- "#7F5757"
database[database$Family=="Moraxellaceae", ]$cols_Family <- "#a73e62"
database[database$Family=="Sphingobacteriaceae", ]$cols_Family <- "#35a4c6"
database[database$Family=="Deinococcaceae", ]$cols_Family <- "#031e33"
database[database$Family=="Weeksellaceae", ]$cols_Family <- "grey"

## collapsed color vector for each level
temp <- data.frame(database$Family, database$cols_Family)
temp <- plyr::ddply(temp, .variables="database.cols_Family", .fun=unique)   
database_level_cols_Family <- as.character(temp[,2])
names(database_level_cols_Family) <- temp[,1]
```

```{r, warning=FALSE, include=FALSE}
# Assignment of colours phylum

database_level_cols_Phylum <- c("Actinobacteria" = "#9A6767", 
                                  "Bacteroidetes" = "#F79869",
                                  "Firmicutes" = "#6799CC",
                                  "Proteobacteria" = "#745B92")

database$cols_Phylum <- database_level_cols_Phylum[database$Phylum]

```


```{r, warning = FALSE, error = FALSE, echo=FALSE, message=FALSE, fig.width=15, fig.height=12}
database_Strain_Family <- database %>% dplyr::select(Strain, Family) %>% filter(Strain %in% strains_tree)

tree_family_branch_line <- ggtree::ggtree(Tree_16s_unique, branch.length = "branch.length", ladderize = TRUE, layout = "circular") %<+% database_Strain_Family +
  geom_tiplab(align=TRUE, linetype='dashed', linesize=.3, size = 3, offset = 0.1) +
  geom_tippoint(aes(color = Family), size = 2) +
  geom_treescale(x=0.7, y=-1.9, fontsize=3, linesize=1, offset=1)+
  scale_color_manual(values = database_level_cols_Family)

# add ring of phylum color between labels and tips
tree_family_branch_phylum <- gheatmap(tree_family_branch_line + new_scale_fill(),
         database %>% 
         dplyr::select(Strain, Phylum ) %>% 
         distinct() %>% na.omit() %>% 
         as.data.frame() %>% column_to_rownames("Strain") ,
         offset=0.02, 
         width=.03,
         colnames = FALSE) +
         scale_fill_manual(values = database_level_cols_Phylum, 
                           labels = c("Actinobacteriota" , 
                                      "Bacteroidota",
                                      "Bacillota" ,
                                      "Pseudomonadota"), 
                           name = "Phylum")
                           
# tree_family_branch_phylum
```

## Fig 1d AMPO formers of MRB collection
The tree is annotated with the quantitative abundance data from mapping greenhouse root microbiome dataset (to B73 WT) and qualitative information on the AMPO formation on the outer ring.

```{r, warning = FALSE, error = FALSE, echo=FALSE, message=FALSE, fig.width=20, fig.height=15}
p_Ch <- gheatmap(tree_family_branch_phylum + new_scale_fill(),
         tree_data %>% 
         # filter(Greenhouse_cum_abundance_99  > 0) %>% 
         dplyr::select(Strain, Greenhouse_cum_abundance_99 ) %>% 
         dplyr::rename(WT_GH = Greenhouse_cum_abundance_99 ) %>% 
         distinct() %>% na.omit() %>% 
         as.data.frame() %>% column_to_rownames("Strain") ,
         offset=0.33, 
         width=.05,
         colnames = FALSE)  +
         scale_fill_gradient2(low = "linen", mid = "gold", high ="coral4", midpoint = 3, 
                              na.value = "white", name="Abundance roots\ngreenhouse")

# unique(tree_data$Morphology_MBOA)

fIG2.pAMPO <- gheatmap(p_Ch + new_scale_fill(),
         tree_data %>% 
         dplyr::select(Strain, Morphology_MBOA) %>% 
         mutate(Morphology_MBOA = gsub("Redmix", "NA", Morphology_MBOA)) %>% 
         distinct() %>% na.omit() %>% 
         as.data.frame() %>% column_to_rownames("Strain") ,
         offset=0.4, 
         width=.05,
         colnames = FALSE,
         colnames_angle= 90, colnames_offset_y = 1, colnames_position="top", font.size=2)  +
         scale_fill_manual(values = c("NR" = "lightgoldenrod", "pale_red"= "tomato1",  "Red"  = "darkred"), 
                           labels = c("no", "weak", "strong"), name="Color on MBOA")


fIG2.pAMPO

ggsave(plot = fIG2.pAMPO,  filename = "Fig1d.svg", width = 25, height = 25, dpi = 300, scale = 1, units = "cm")
```

## Fig S4 Abundance of AMPO formers in microbiome of field data 
Abundance of isolates from strain collection in microbiome datasets of the field and greenhouse. Mapping data and relative abundances of ASVs are taken from Thoenen et al 2023 PNAS.

Calculate cumulative abundance (after filtering of ASVs for combining) in greenhouse B73 WT (Hu Feedback, 2018)
```{r, error = FALSE, echo=FALSE, message=FALSE}
# filter all ASVs mapping to at least one AMPO former
AMPOformers <- database %>% 
  filter(Morphology_MBOA %in% c("Red", "pale_red")) %>% 
  pull(Strain)

ASVs_Hu_Feedback_AMPO <- ASVs_Hu_Feedback %>% filter(Strain %in% AMPOformers) %>% 
  filter(X. > cutoff_similarity) %>% 
  dplyr::select(OTU, phylum, family,soil, Abundace_mean) %>% 
  unique() %>% # only keep one copy for each ASV 
  group_by(phylum, family) %>% 
  summarise(Greenhouse_cum_abundance = sum(Abundace_mean))

ASVs_Hu_Feedback_AMPO_cum_abund <- ASVs_Hu_Feedback_AMPO %>%  
  mutate(location_background = "Greenhouse_B73", compartment ="root", cum_abundance = Greenhouse_cum_abundance) %>% 
    dplyr::select(phylum, family, location_background, compartment, cum_abundance) 

```

Calculate cumulative abundance (after filtering of ASVs for combining) in field data for Changins, Reckenholz and Aurora (Cadot et al, 2021). Mappings taken from Thoenen et al 2023.
```{r, warning = FALSE, error = FALSE, echo=FALSE, message=FALSE, fig.width=10, fig.height=7.5}
# read in relative abundance data of Cadot et al
ASVs_Cadot <- read_rds("Input/all_phy_psmelt_Cadot_iso.rds")  

# get list of all ASVs that contain at least one AMPO former
AMPO_ASVs_Cadot <- ASVs_Cadot %>% 
  filter(Strain %in% AMPOformers) %>% 
  filter(`%` > cutoff_similarity) %>% 
  pull(OTU) %>% unique()

# keep only those ASVs and only one unique row per ASV
AMPO_ASV_abundance_Cadot_cum_abund <- ASVs_Cadot %>% 
  filter(OTU %in% AMPO_ASVs_Cadot) %>% 
  filter(compartment == "root" & background %in% c("B73", "W22") & genotype == "WT") %>% 
  mutate(location = gsub("Zurich", "Reckenholz", location)) %>% 
  mutate(location = gsub("Ithaca", "Aurora", location)) %>% 
  ungroup() %>% 
  dplyr::select(OTU, phylum, family, location, background, compartment, Abundance_mean = Abundace_mean) %>% 
  mutate(location_background = interaction(location, background, sep = "_")) %>% 
  filter(location_background %in% c("Changins_B73", "Reckenholz_B73", "Aurora_W22")) %>%
  dplyr::arrange(desc(Abundance_mean)) %>% 
  unique() %>% 
  group_by(phylum, family, location_background, compartment) %>% 
  summarise(cum_abundance = sum(Abundance_mean ))

```

```{r, warning = FALSE, error = FALSE, echo=FALSE, message=FALSE, fig.width=10, fig.height=7.5}

#combine Greenhouse (Hu et al) and field data 
AMPO_ASV_abundance_combined <- rbind(ASVs_Hu_Feedback_AMPO_cum_abund, AMPO_ASV_abundance_Cadot_cum_abund) 
  
AMPO_in_fields_ASVs <- AMPO_ASV_abundance_combined %>% 
  mutate(Location = location_background) %>% 
  separate(Location, into = c("Location", "genotype")) %>% 
  mutate(Location = factor(Location, levels = c("Greenhouse", "Changins", "Reckenholz", "Aurora"))) %>%
  na.omit() %>%
  ggplot(aes(x = Location, y = cum_abundance)) +
  geom_bar(aes(fill = family), stat = "summary") +
   theme_classic() +
  theme(axis.text.x=element_text(angle = -90, hjust = 0, vjust = 0.5))+
  theme(plot.title = element_text(size = 28/.pt)) +
  theme(axis.text.x = element_text(color="black"),
    axis.ticks = element_line(color = "black"))+
  scale_fill_manual(values = database_level_cols_Family) +
  labs(y = "Abundance [%]", x = "", title = "AMPO genera mapped to microbiomes")

AMPO_in_fields_ASVs

ggsave(plot = AMPO_in_fields_ASVs, filename = paste0("FigS4.svg"), width = 8.8, height = 9, dpi = 300, scale = 1, units = "cm")

```


```{r, warning = FALSE, error = FALSE, echo=FALSE, message=FALSE, fig.width=10, fig.height=7.5}

# table per location, note all WT
AMPO_ASV_abundance_combined %>% knitr::kable()

#table for total mapped ASVs of both Cadot and Hu et al
AMPO_ASV_abundance_combined %>% 
  mutate(Location = location_background) %>% 
  separate(Location, into = c("Location", "genotype")) %>% 
  mutate(Location = factor(Location, levels = c("Greenhouse", "Changins", "Reckenholz", "Aurora"))) %>%
  na.omit() %>%
  ungroup() %>% 
  group_by(Location, genotype) %>% 
  summarise(cumulative_abundance = sum(cum_abundance))  %>% 
  knitr::kable()

```

number of strains tested and number of AMPO formers
```{r, warning=FALSE, error = FALSE, echo=FALSE, message=FALSE}
# summary(factor(tree_data$Morphology_MBOA))

summary_Red <- tree_data %>% select(Strain, Phylum, Class, Order, Family, Genus, Morphology_MBOA) %>% 
  group_by(Phylum, Class, Order, Family, Genus, Morphology_MBOA) %>% 
  summarise(n()) 

summary_Red %>% knitr::kable()

```



```{r}
sessionInfo()
```


